File: problems/338-Counting-Bits/solution.go
func countBits(n int) []int {
	result := make([]int, n+1)
	for i := 0; i <= n; i++ {
		num := i
		for num > 0 {
			if num&0b1 == 1 {
				result[i]++
			}
			num >>= 1
		}
	}
	return result
}

File: problems/338-Counting-Bits/solution.java
class Solution {
    private int count(int n) {
        int result = 0;
        while (n > 0) {
            if ((n & 1) == 1) {
                result++;
            }

            n >>= 1;
        }

        return result;
    }

    public int[] countBits(int n) {
        int[] result = new int[n + 1];
        for (int i = 0; i <= n; i++) {
            result[i] = count(i);
        }

        return result;
    }
}

